### PR TITLE
Hhy revised

### PR DESCRIPTION
**Pull request by Mongolialady**
#### Pre-Submission Checklist:

- [ ] Your pull request targets the `master` branch of calltocode.org
- [ ] Your branch is updated with the latest changes on `master`
- [ ] Your branch passes linting, building and all new and existing tests with command `yarn test`
- [ ] The story that this pull request addresses is referenced above by ID

#### Post-Submission Checklist:

- [ ] Address any reviews on your pull request, as it will not be merged until it is approved
- [ ] Your branch has no conflicts with `master` (find this out at the bottom of your pull request)

### CODE REVIEW

**Comment by Mongolialady** [comment]
@sexta13 
Yes. The story is https://www.pivotaltracker.com/story/show/153556693
Login button should get enabled when there is at least one character in the password field

File: client/actions/index.js [hunk 1]
```diff
@@ -1,3 +1,4 @@
 export { default as AuthActionCreator } from './auth'
 export { default as ProjectActionCreator } from './project'
 export { default as UserActionCreator } from './user'
+export { default as headerActionCreator } from './header'
```
Comment (bbmoz) [inline client/actions/index.js]: Use upper-case `HeaderActionCreator`

File: client/components/Header/Header.js [hunk 2]
```diff
@@ -20,46 +18,30 @@ class Header extends Component {
   }
 
   renderHeaderButtons () {
-    const { authenticated, user } = this.props
+    var {login} = this.props
```
Comment (bbmoz) [inline client/components/Header/Header.js]: Opt to use `const` wherever you can.

File: client/components/Header/Header.js [hunk 3]
```diff
@@ -20,46 +18,30 @@ class Header extends Component {
...
+    console.log('had been logged', login)
```
Comment (bbmoz) [inline client/components/Header/Header.js]: Strip `console.log`'s before opening a PR.

File: client/components/Header/Header.js [hunk 4]
```diff
@@ -20,46 +18,30 @@ class Header extends Component {
...
-            {'SIGN UP'}
-          </Button>
-          <Button dense className={this.getLinkStyles('login')}
-            component={Link}
-            to="/login">
-            {'LOG IN'}
-          </Button>
-        </Fragment>
-      )
+      var buttons = [<Link key="signup" to="/signup" className={this.getLinkStyles('signup')}>SIGN UP</Link>]
```
Comment (bbmoz) [inline client/components/Header/Header.js]: Opt for `const` here.

File: client/components/LoginForm/LoginForm.js [hunk 5]
```diff
@@ -45,30 +49,53 @@ class LoginForm extends Component {
...
+    else {
+      enableLogin();
+    }
   }
 
-  render () {
-    const { classes } = this.props
+
+  componentWillUnmount() {
+    var {enableLogin} = this.props;
```
Comment (bbmoz) [inline client/components/LoginForm/LoginForm.js]: Use `const` here

File: client/components/LoginForm/LoginForm.js [hunk 6]
```diff
@@ -45,30 +49,53 @@ class LoginForm extends Component {
...
-    this.setState({ [event.target.name]: event.target.value })
+  handleChange(event) {
+    event.preventDefault();
+    this.setState({[event.target.name]: event.target.value})
+  }
+
+  enableOrDisableLogin() {
+    var email = this.state.email || '';
+    var password = this.state.password || '';
+    var {enableLogin, disableLogin} = this.props;
```
Comment (bbmoz) [inline client/components/LoginForm/LoginForm.js]: Opt for `const`

File: client/components/LoginForm/LoginForm.js [hunk 7]
```diff
@@ -45,30 +49,53 @@ class LoginForm extends Component {
     }
   }
 
-  onBlur (event) {
-    event.preventDefault()
+  onBlur(event) {
+    event.preventDefault();
```
Comment (bbmoz) [inline client/components/LoginForm/LoginForm.js]: Remove semicolons, following the linting guidelines for standardjs, unless the semicolon is absolutely required

File: client/actions/header/index.js [hunk 8]
```diff
@@ -0,0 +1,20 @@
+/**
+ * Created by hhy615 on 1/19/18.
+ */
```
Comment (bbmoz) [inline client/actions/header/index.js]: Strip comments before a PR

File: client/actions/header/types.js [hunk 9]
```diff
@@ -0,0 +1,5 @@
+/**
+ * Created by hhy615 on 1/19/18.
+ */
```
Comment (bbmoz) [inline client/actions/header/types.js]: Strip comment

File: server/config/database.js [hunk 10]
```diff
@@ -1,5 +1,8 @@
 export default {
   get url () {
+
+    process.env.NODE_ENV=process.env.NODE_ENV|| "dev";
```
Comment (bbmoz) [inline server/config/database.js]: Instead of overwriting `process.env.NODE_ENV`, try to use the switch `default` case instead.

**Comment by bbmoz** [comment]
In addition, please fix your PR title as well as your branch following these guidelines:
- [branching](https://github.com/CodeForSocialGood/calltocode.org/blob/master/docs/CONTRIBUTING.md#branching)
- [pull request](https://github.com/CodeForSocialGood/calltocode.org/blob/master/docs/CONTRIBUTING.md#-pull-request)

**Pull request closed without merging**